birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+These hooks will be running within Twitcher, using MagpieAdapter context, applied for STAC requests.
+
+The code below can make use of any package that is installed by Magpie/Twitcher.
+
+.. seealso::
+    Documentation about Magpie/Twitcher request/response hooks is available here:
+    https://pavics-magpie.readthedocs.io/en/latest/configuration.html#service-hooks
+"""
+
+import json
+import re
+from typing import TYPE_CHECKING
+
+from magpie.api.management.resource import resource_utils as ru
+from magpie.api.requests import get_service_matchdict_checked
+from magpie.models import Route, Service
+from magpie.utils import get_logger
+from magpie.db import get_session_from_other
+from ziggurat_foundations.models.services.resource import ResourceService
+
+if TYPE_CHECKING:
+    from pyramid.request import Request
+    from pyramid.response import Response
+
+LOGGER = get_logger("magpie.stac")
+
+def create_collection_resource(response):
+    # type: (Response) -> Response
+    """
+    Create the stac collection resource
+    """
+    request = response.request
+    body = request.json
+    collection_id = body["id"]
+    try:
+        display_name = extract_display_name(body["links"])
+    except Exception as exc:
+        LOGGER.error("Error when extracting display_name from links %s %s", body["links"], str(exc), exc_info=exc)
+        return response
+    # note: matchdict reference of Twitcher owsproxy view is used, just so happens to be same name as Magpie
+    service = get_service_matchdict_checked(request)
+    try:
+        # Getting a new session from the request, since the current session found in the request is already handled with his own transaction manager.
+        session = get_session_from_other(request.db)
+        # Create the resource tree
+        create_resource_tree(f"stac/collections/{collection_id}", 0, service.resource_id , session, display_name)
+        session.commit()
+
+    except Exception as exc:
+        LOGGER.error("Unexpected error while creating the collection %s %s", display_name, str(exc), exc_info=exc)
+        session.rollback() 
+        return response
+
+    return response
+
+def create_item_resource(response):
+    # type: (Response) -> Response
+    """
+    Create the stac item resource
+    """
+    request = response.request
+    body = request.json
+    item_id = body["id"]
+    try:
+        display_name = extract_display_name(body["links"])
+    except Exception as exc:
+        LOGGER.error("Error when extracting display_name from links %s %s", body["links"], str(exc), exc_info=exc)
+        return response
+
+    # Get the <collection_id> from url -> /collections/{collection_id}/items
+    collection_id = re.search(r'(?<=collections\/).*?(?=\/items)', request.url).group()
```

Comment:
The regex should be improved to use accepted characters (`[0-9a-zA-Z_.-]+`) instead of `.*?`.
Currently, some invalid regexes are mached such as `/collections//items` or `/collections/   /items`, which will cause an error when reaching the DB queries.